### PR TITLE
feat: expose allowed_ip_addresses in the solution as a variable

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
root level module already passes variable through to the iam account settings module, this change exposes that as a variable w/ default at the solution layer
also update common-dev-assets version which contains fixes for checkov on mac failing

resolves [terraform-ibm-account-infrastructure-base/issues#51](https://github.com/terraform-ibm-modules/terraform-ibm-account-infrastructure-base/issues/51)

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

expose var.allowed_ip_addresses in the solution

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: solutions/account-infrastructure-base/variables.tf [hunk 1]
```diff
@@ -39,3 +39,9 @@ variable "cos_bucket_management_endpoint_type" {
     error_message = "The specified management_endpoint_type_for_bucket is not a valid selection!"
   }
 }
+
+variable "allowed_ip_addresses" {
+  description = "List of the IP addresses and subnets from which IAM tokens can be created for the account."
+  type        = list(string)
+  default     = ["0.0.0.0/0"]
```
Comment (ocofaigh) [inline solutions/account-infrastructure-base/variables.tf:46]: Do not default to this! Intact this should be a required variable, so remove all defaults. We can pass this value in the TEST, no here

Comment (ocofaigh) [reply solutions/account-infrastructure-base/variables.tf:46]: If its required, you will also need to add it to the cra-config.yaml and catalogValidationValues.json.template

Comment (MatthewLemmond) [reply solutions/account-infrastructure-base/variables.tf]: I've updated the variable to no longer have a default and passed it in to the test/cra/catalog validation, I wasn't sure on formatting for a list when passing as a variable in the yaml so I left it as a string but if it needs to be changed to a yaml list I can update it

File: solutions/account-infrastructure-base/catalogValidationValues.json.template [hunk 2]
```diff
@@ -3,5 +3,6 @@
   "region": "us-south",
   "resource_tags": $TAGS,
   "prefix": $PREFIX,
-  "kms_key_crn": $HPCS_US_SOUTH_ROOT_KEY_CRN
+  "kms_key_crn": $HPCS_US_SOUTH_ROOT_KEY_CRN,
+  "allowed_ip_addressses": ["0.0.0.0/0"]
```
Comment (ocofaigh) [inline solutions/account-infrastructure-base/catalogValidationValues.json.template:7]: This is json, so needs to be `"[\"0.0.0.0/0\"]"`

File: cra-config.yaml [hunk 3]
```diff
@@ -10,3 +10,4 @@ CRA_TARGETS:
     CRA_ENVIRONMENT_VARIABLES:  # An optional map of environment variables for CRA, where the key is the variable name and value is the value. Useful for providing TF_VARs.
       TF_VAR_kms_key_crn: "crn:v1:bluemix:public:hs-crypto:us-south:a/abac0df06b644a9cabc6e44f55b3880e:e6dce284-e80f-46e1-a3c1-830f7adff7a9:key:1368d2eb-3ed0-4a8b-b09c-2155895f01ea"
       TF_VAR_prefix: "test"
+      TF_VAR_allowed_ip_addresses: "[\"0.0.0.0/0\"]"
```
Comment (ocofaigh) [inline cra-config.yaml:13]: actually, for CRA we will want to use some locked down IP (as this means allow all, so might fail scan some day). Set it to `"[\"17.5.7.8.0/16\"]"` for example - it wont actually apply this for CRA scan, just does a plan

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**